ihmc-perception/src/main/java/us/ihmc/sensors/ZEDImageGrabber.java

```diff
@@ -0,0 +1,325 @@
+      zedRuntimeParameters.reference_frame(SL_REFERENCE_FRAME_CAMERA);
+      zedRuntimeParameters.enable_depth(true);
+      zedRuntimeParameters.confidence_threshold(100);
+      zedRuntimeParameters.texture_confidence_threshold(100);
+      zedRuntimeParameters.remove_saturated_areas(true);
+      zedRuntimeParameters.enable_fill_mode(false);
+   }
+
+   @Override
+   public boolean startSensor()
+   {
+      try
+      {
+         sl_create_camera(cameraID);
+
+         // Set the initialization parameters
+         setInitParameters(zedInitParameters);
+
+         // Open the camera
+         int returnCode = sl_open_camera(cameraID, zedInitParameters, 0, "", "", 0, "", "", "");
+         throwOnError(returnCode);
+
+         // Get camera intrinsics
+         SL_CalibrationParameters sensorIntrinsics = sl_get_calibration_parameters(cameraID, false);
+         imageWidth = sl_get_width(cameraID);
+         imageHeight = sl_get_height(cameraID);
+
+         leftSensorIntrinsics.setWidth(imageWidth);
+         leftSensorIntrinsics.setHeight(imageHeight);
+         leftSensorIntrinsics.setFx(sensorIntrinsics.left_cam().fx());
+         leftSensorIntrinsics.setFy(sensorIntrinsics.left_cam().fy());
+         leftSensorIntrinsics.setCx(sensorIntrinsics.left_cam().cx());
+         leftSensorIntrinsics.setCy(sensorIntrinsics.left_cam().cy());
+
+         rightSensorIntrinsics.setWidth(imageWidth);
+         rightSensorIntrinsics.setHeight(imageHeight);
+         rightSensorIntrinsics.setFx(sensorIntrinsics.right_cam().fx());
+         rightSensorIntrinsics.setFy(sensorIntrinsics.right_cam().fy());
+         rightSensorIntrinsics.setCx(sensorIntrinsics.right_cam().cx());
+         rightSensorIntrinsics.setCy(sensorIntrinsics.right_cam().cy());
+         sensorIntrinsics.close();
+
+         // Get center to camera distance
+         SL_CameraInformation cameraInformation = sl_get_camera_information(cameraID, 0, 0);
+         sensorCenterToCameraDistanceY = 0.5f * cameraInformation.camera_configuration().calibration_parameters().translation().y();
+         cameraInformation.close();
+
+         // Create image retrieval pointers
+         slMatPointers[LEFT_COLOR_IMAGE_KEY] = new Pointer(sl_mat_create_new(imageWidth, imageHeight, SL_MAT_TYPE_U8_C4, SL_MEM_GPU));
+         slMatPointers[RIGHT_COLOR_IMAGE_KEY] = new Pointer(sl_mat_create_new(imageWidth, imageHeight, SL_MAT_TYPE_U8_C4, SL_MEM_GPU));
+         slMatPointers[DEPTH_IMAGE_KEY] = new Pointer(sl_mat_create_new(imageWidth, imageHeight, SL_MAT_TYPE_U16_C1, SL_MEM_GPU));
+      }
+      catch (ZEDException exception)
+      {
+         LogTools.error(exception);
+         return false;
+      }
+
+      lastGrabFailed = false;
+      return true;
+   }
+
+   protected void setInitParameters(SL_InitParameters parametersToSet)
+   {
+      parametersToSet.camera_fps(CAMERA_FPS);
+      parametersToSet.resolution(SL_RESOLUTION_HD720);
+      parametersToSet.input_type(SL_INPUT_TYPE_USB);
```

Comment:
How can we handle GMSL?

Reply:
Good question haha. I'm debating between adding a constructor parameters and passing in the whole init parameters... Will figure it out a bit later.